### DISCUSSION TITLE
flinkcdc用API方式读取mongodb的多个数据源报错？单个数据源没有问题，就是报这个错误org.bson.conversions.Bson.toBsonDocument()Lorg/bson/BsonDocument;

### DISCUSSION THREAD

**Discussion by tuoluzhe8521** [apache/flink-cdc#1609] [Q&A]
为什么flinkcdc用API方式读取mongodb的多个数据源会报错呢？就是报这个错误org.bson.conversions.Bson.toBsonDocument()Lorg/bson/BsonDocument有大神知道么？下面是我粘贴的代码。
报错信息为：

`ERROR (com.ververica.cdc.debezium.internal.Handover:reportError) - Reporting error:
java.lang.NoSuchMethodError: org.bson.conversions.Bson.toBsonDocument()Lorg/bson/BsonDocument;`

代码：
`
StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
       env.setParallelism(1);
        env.setBufferTimeout(BUFFER_TIMEOUT_MS);
        env.enableCheckpointing(CHECKPOINT_INTERVAL_MS, CheckpointingMode.AT_LEAST_ONCE);
        env.getCheckpointConfig().setCheckpointTimeout(CHECKPOINT_TIMEOUT_MS);
        env.getCheckpointConfig().setMinPauseBetweenCheckpoints(CHECKPOINT_MIN_PAUSE_MS);
        env.setRestartStrategy(RestartStrategies.failureRateRestart(3, Time.minutes(1), Time.seconds(10)));
StreamTableEnvironment.create(env);

        //2.通过FlinkCDC构建SourceFunction
        SourceFunction<String> mongoDBSourceFunction = MongoDBSource.<String>builder()
                .hosts(MONGODB_URL)
                .username(MONGODB_USER)
                .password(MONGODB_PWD)
                .databaseList(MONGODB_DATABASE)
       //此处读取两个表报错，读取一个表没问题         .collectionList("thread_test.user_test1","thread_test.user_test2")
               .deserializer(new JsonDebeziumDeserializationSchema())
                .build();
   

        DataStreamSink<String> mongo_to_hdfs = env.addSource(mongoDBSourceFunction)
                .name("mongo_to_hdfs")
                .print().setParallelism(1);
        env.execute();
`

**Comment by EMsnap** [comment]
seems like there are loss of some jar files, or some jars are conflicted
